### PR TITLE
Update app when new docker hub credentials are create/updated

### PR DESCRIPTION
**Pull request by divolgin**
#### What type of PR is this?

type::bug
#### What this PR does / why we need it:

When DockerHub credentials are created/update, this will create a new app version with the new pull secret.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
* Fixes an issue that caused instances to run into DockerHub's rate limiting after running the [ensure-secret](/reference/kots-cli-docker-ensure-secret) command
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: pkg/midstream/write.go [hunk 1]
```diff
@@ -186,6 +188,7 @@ func (m *Midstream) writeDisasterRecoveryLabelTransformer(options WriteOptions)
 }
 
 func (m *Midstream) writePullSecret(options WriteOptions) (string, error) {
+	debug.PrintStack()
```
Comment (sgalsaleh) [inline pkg/midstream/write.go:191]: cleanup?

File: pkg/template/config_context.go [hunk 2]
```diff
@@ -374,6 +374,11 @@ func (ctx ConfigCtx) localRegistryImagePullSecret() string {
 		}
 		secret = secrets.AppSecret
 	}
+
+	// if secret == nil {
```
Comment (sgalsaleh) [inline pkg/template/config_context.go:378]: ?

**Review by sgalsaleh — DISMISSED** [review]

File: .github/workflows/build-test.yaml [hunk 3]
```diff
@@ -1109,6 +1109,20 @@ jobs:
           printf "App is installed successfully with the correct version label: $APP_VERSION_LABEL\n\n"
           ./bin/kots get apps --namespace $APP_SLUG
 
+          # test setting DockerHub credentials
```
Comment (divolgin) [inline .github/workflows/build-test.yaml:1112]: this is new

File: pkg/handlers/config.go [hunk 4]
```diff
@@ -487,25 +487,25 @@ func updateAppConfig(updateApp *apptypes.App, sequence int64, configGroups []kot
 
 	// we don't merge, this is a wholesale replacement of the config values
 	// so we don't need the complex logic in kots, we can just write
-	values := kotsKinds.ConfigValues.Spec.Values
```
Comment (divolgin) [inline pkg/handlers/config.go:490]: this is new

File: pkg/image/builder.go [hunk 5]
```diff
@@ -94,7 +94,7 @@ func ProcessImages(srcRegistry, destRegistry registry.RegistryOptions, appSlug s
 func GetPrivateImages(upstreamDir string, checkedImages map[string]ImageInfo, allPrivate bool, dockerHubRegistry registry.RegistryOptions, useHelmInstall map[string]bool) ([]string, []k8sdoc.K8sDoc, error) {
 	uniqueImages := make(map[string]bool)
 
-	objects := make([]k8sdoc.K8sDoc, 0) // all objects where images are referenced from
```
Comment (divolgin) [inline pkg/image/builder.go:97]: this is new

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**